resources/puppetlabs/lein-ezbake/template/global/ext/redhat/preinst.erb

```diff
@@ -2,15 +2,24 @@
 # install.sh source based installation methodology.
 #
 # Add <%= EZBake::Config[:group] %> group
-getent group <%= EZBake::Config[:group] %> > /dev/null || \
-  groupadd -r <%= EZBake::Config[:group] %> || :
+<% if EZBake::Config[:numeric_uid_gid].nil? -%>
+getent group <%= EZBake::Config[:group] %> >/dev/null || groupadd --system --force <%= EZBake::Config[:group] %>
+<% else -%>
+getent group <%= EZBake::Config[:group] %> >/dev/null || groupadd --system --force --gid <%= EZBake::Config[:numeric_uid_gid] %> <%= EZBake::Config[:group] %>
+<% end -%>
+
 # Add <%= EZBake::Config[:user] %> user
 if getent passwd <%= EZBake::Config[:user] %> > /dev/null; then
   usermod --gid <%= EZBake::Config[:group] %> --home %{_app_data} \
   --comment "<%= EZBake::Config[:project] %> daemon" <%= EZBake::Config[:user] %> || :
 else
-  useradd -r --gid <%= EZBake::Config[:group] %> --home %{_app_data} --shell $(which nologin) \
-    --comment "<%= EZBake::Config[:project] %> daemon"  <%= EZBake::Config[:user] %> || :
+  useradd_options=('--system' '--gid' '<%= EZBake::Config[:group] %>' '--home' '%{_app_data}' '--shell' "$(which nologin)" '--comment' '<%= EZBake::Config[:project] %> daemon')
+<% unless EZBake::Config[:numeric_uid_gid].nil? -%>
+  if ! getent passwd <%= EZBake::Config[:numeric_uid_gid] %> > /dev/null; then
+    useradd_options+=('--uid' '<%= EZBake::Config[:numeric_uid_gid] %>')
+  fi
+<% end -%>
+  useradd "${useradd_options[@]}" <%= EZBake::Config[:user] %> || :
```

Comment:
I neither love nor hate this change, what do y'alls think 🤷‍♀️

Reply:
I think I like this OK, honestly?

Reply:
^same.

Can you explain why you have to check if the uid/gid exists already when adding a user, but not when adding a group?

Reply:
We're running groupadd with `--force`, so if we try to create it with a gid that already exists it will pick a different one ([manpage](https://linux.die.net/man/8/groupadd)). `useradd` doesn't support that functionality.

Reply:
cool, thanks!